src/rcore.c

```diff
@@ -1946,26 +1946,23 @@ const char *GetFileName(const char *filePath)
 // Get filename string without extension (uses static string)
 const char *GetFileNameWithoutExt(const char *filePath)
 {
-    #define MAX_FILENAMEWITHOUTEXT_LENGTH   256
-
-    static char fileName[MAX_FILENAMEWITHOUTEXT_LENGTH] = { 0 };
-    memset(fileName, 0, MAX_FILENAMEWITHOUTEXT_LENGTH);
-
-    if (filePath != NULL) strcpy(fileName, GetFileName(filePath));   // Get filename with extension
-
-    int size = (int)strlen(fileName);   // Get size in bytes
-
-    for (int i = 0; (i < size) && (i < MAX_FILENAMEWITHOUTEXT_LENGTH); i++)
+    if (filePath != NULL)
     {
-        if (fileName[i] == '.')
+        const char *fileName = GetFileName(filePath); // Get filename.ext without path
+        for (int i = (int)strlen(fileName); i>0; i--) // Reverse search '.'
         {
-            // NOTE: We break on first '.' found
-            fileName[i] = '\0';
-            break;
+            if (fileName[i] == '.')
+            {
+                char rfileName[i];
+                strncpy(rfileName, fileName,i);
+                rfileName[i] = '\0';
+                const char* rrfileName = rfileName;
+                return rrfileName;
+            }
         }
+        return fileName; // Extension not found
```

Comment:
In raylib I tried to avoid, as much as possible, any early return. I prefer to assign required value and wait to the end of the function to return.

Reply:
Ok i understand why you do that, i back with fix.